ckan/controllers/user.py

```diff
@@ -80,7 +80,8 @@ def _setup_template_variables(self, context, data_dict):
         try:
             user_dict = get_action('user_show')(context, data_dict)
         except NotFound:
-            abort(404, _('User not found'))
+            h.flash_error('Authorize to see this page')
+            h.redirect_to(controller='user', action='login')
```

Comment:
This doesn't seem to distinguish between a missing authentication and a user that genuinely doesn't exist. If `get_action('user_show')` raises `NotFound`, then a 404 seems like the right response. If that's not the case in the situtation you're trying to fix then perhaps `user_show` shouldn't raise `NotFound` in the first place, but `NotAuthorized` (or, if we're just requiring that the user is logged in, a currently not existing `NotAuthenticated`).